src/spikeinterface/core/baserecording.py

```diff
@@ -458,12 +458,26 @@ def has_time_vector(self, segment_index=None):
         return d["time_vector"] is not None
 
     def set_times(self, times, segment_index=None, with_warning=True):
-        """Set times for a recording segment.
+        """Set times for a recording segment. Any existing times
+        will be overwritten.
+
+        Times can be manually set on the recording segment. If times are
+        not set, the sample index and sampling frequency are used to
+        calculate time. Otherwise, `t_start` or `time_vector` can be
+        provided:
+
+        `t_start` - the start time for the segment. The times for
+            this recording segment will be calculated as
+            t_start + sample_index * (1 / sampling_frequency)
+
+        `time_vector` - A vector of length segment.get_num_samples()
+            that holds the exact time for each sample in the recording.
 
         Parameters
         ----------
-        times : 1d np.array
-            The time vector
+        times : int | float | 1d np.array
```

Comment:
I personally would prefer to not overload this function and create a new one instead. `set_t_start`.

 What are the advantages of overloading this? How are you thinking about it?

Reply:
Yes I couldn't decide on this vs. separate functions and @samuelgarcia suggested this approach. On balance I think I prefer it as `t_start` and `times_vector` are mutually exclusive ways of setting custom times, so it makes sense to change in one place. It would be slightly strange to call `set_t_start()` and this removes the `time_vector` attribute. But I'm not sure either way.

Reply:
Interesting. Thanks for sharing, I will add this as something to discuss at the meeting.

Reply:
I guess that my take is the following:

We use `set_times` because the sampling recording is _slightly_ irregular when we want more precision. Why we would need to set `t_start`. 

The use case that I can think off is that we would like to shift all the recording to the right or the left on time. But if that is the use case I would be better to have a method that shift the recording times and works independently of whether times are handled internally with t_start and sampling frequency or a time vector.

If it is not for shifting I can't thik on other use of setting `t_start`.

Reply:
I think the use case would be if you had separate sessions in a single day, for example a session, 10 minute break to change some equiptment, and another recording session. If these sessions are held as different segments on a recording (or, as separate recordings) the researcher may want to hold the true recording times (e.g. session 1 started at 1 pm, session two at 1.30 pm).

Reply:
OK, change my answer below what I think this type of case is not well served with the current implementation.